wolox-books/src/app/signup-form/signup-form.component.ts

```diff
@@ -1,15 +1,46 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+const SECONDARY_BUTTON_TEXT = 'Log In';
 
 @Component({
   selector: 'app-signup-form',
   templateUrl: './signup-form.component.html',
   styleUrls: ['./signup-form.component.scss']
 })
+
 export class SignupFormComponent implements OnInit {
 
-  constructor() { }
+  constructor(private formBuilder: FormBuilder) { 
+
+  }
+
+  secondaryButtonText: string = SECONDARY_BUTTON_TEXT;
+  signUpForm: FormGroup;
 
   ngOnInit(): void {
+    this.signUpForm = this.formBuilder.group(
+      {
+        name: '',
+        lastName: '',
+        email: ['', [Validators.required, Validators.email]],
+        password: ['', [Validators.required, Validators.pattern('^(?=.*[0-9])(?=.*[A-Z]).*$')]],
```

Comment:
I would store the password regex on another file (validators.ts) in case we need to reuse it somewhere